### ISSUE TITLE
linter: `no-redeclare`: `should_run` disables the entire rule for ES modules, not just builtin-global checks

### ISSUE THREAD

**Issue by rbertozg** [oxc-project/oxc#25339]
### What version of Oxlint are you using?

1.76.0

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
// Minimal repro config
{
  "categories": {
    "correctness": "off"
  },
  "rules": {
    "no-redeclare": "error"
  }
}
```

### What happened?

`no-redeclare` never reports anything on regular `.ts`/`.tsx` files, including basic cases like:

```ts
var a = 3;
var a = 10;
```

and the TS-specific type/value collision case covered by #10396:

```ts
type foo = 1;
export function foo(): void;
export function foo() { }
```

Root cause, in `crates/oxc_linter/src/rules/eslint/no_redeclare.rs`:

```rust
fn should_run(&self, ctx: &ContextHost) -> bool {
    // ES modules run in their own scope, and don't conflict with existing globals
    !ctx.source_type().is_module()
}
```

`ModuleKind::Module` is the **default** `module_kind` for `.ts`/`.tsx`/`.mts` files (per `oxc_span::SourceType`), so for a normal ESM TypeScript project this rule is effectively dead code — `should_run` returns `false` for essentially every real source file. It only fires on `.cts` (explicitly CommonJS) or `Script`-kind files. This is also why the rule's own tests need `.change_rule_path_extension(".cts")` to exercise the logic at all — on the default extension, the tests would pass trivially without the rule ever running.

The comment justifies this only for **builtin-global** redeclarations ("ES modules run in their own scope, and don't conflict with existing globals" — which correctly explains why an ESM file shouldn't be flagged for shadowing a builtin `Object`/`self`/`globalThis` etc., since each module has its own scope). But the `should_run` gate disables the **entire rule**, including:
- local `var`/`let`/`const` redeclarations within the same file (`var a; var a;`), which have nothing to do with global scope,
- the TS-specific type/value name collision check added for #10396.

### What did you expect to happen?

Local redeclaration checks (and the type/value collision check) should still run in ES modules — only the *builtin-global* branch should be skipped for `is_module()`. Gating just the `builtin_globals` check on `!is_module()`, while always running the local-symbol redeclaration loop, would match the stated rationale in the comment.

### Additional context

Since ESM is the standard module system for modern TS/React/Node projects, `no-redeclare` is currently a no-op for the overwhelming majority of real-world oxlint users who enable it, with no warning or indication that it isn't running.
